### PR TITLE
[tensorflow] [build] [test] Updating TF binaries for 1.15.4 CUDA10.1 PY36 and PY37

### PR DESCRIPTION
**Pull request by Teng-xu**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Update tf binaries for tf 1.15.4 cuda10.1 with py36 and py37
*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/training/docker/1.15.4/py36/cu101/Dockerfile.gpu [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+# Specify the location of module that contains the training logic for SageMaker
+# https://docs.aws.amazon.com/sagemaker/latest/dg/docker-container-environmental-variables-entrypoint.html
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_tensorflow_container.training:main
+
+# Define framework-related package sources
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r1.15_aws/20201002-042937/gpu/tensorflow_gpu-1.15.4-cp36-cp36m-manylinux2010_x86_64.whl
+
+# ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/estimator/r1.15_aws/20201001-204030/tensorflow_estimator-1.15.2-py2.py3-none-any.whl
```
Comment (Teng-xu) [inline tensorflow/training/docker/1.15.4/py36/cu101/Dockerfile.gpu:23]: In the previous docker files 1.15.3, in both cpu and gpu images don't install estimator wheel, so I want to confirm that do we need to install estimator for 1.15.4

File: tensorflow/training/docker/1.15.4/py37/cu101/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+ENV PYTHONIOENCODING=UTF-8
+ENV LANG=C.UTF-8
+ENV LC_ALL=C.UTF-8
+# Specify the location of module that contains the training logic for SageMaker
+# https://docs.aws.amazon.com/sagemaker/latest/dg/docker-container-environmental-variables-entrypoint.html
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_tensorflow_container.training:main
+
+# Define framework-related package sources
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r1.15_aws/20201002-042937/gpu/py37/tensorflow_gpu-1.15.4-cp37-cp37m-manylinux2010_x86_64.whl
+# ARG ESTIMATOR_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/estimator/r1.15_aws/20201001-204030/tensorflow_estimator-1.15.2-py2.py3-none-any.whl
```
Comment (Teng-xu) [inline tensorflow/training/docker/1.15.4/py37/cu101/Dockerfile.gpu:22]: Same question here, do we need to install estimator for 1.15.4 images?

File: tensorflow/buildspec-tf1.yml [hunk 3]
```diff
@@ -37,9 +37,9 @@
     BuildTensorflowCPUTrainPy3DockerImage:
       <<: *TRAINING_REPOSITORY
       build: &TENSORFLOW_CPU_TRAINING_PY3 false
-      image_size_baseline: 4899
+      image_size_baseline: 5975
       device_type: &DEVICE_TYPE cpu
-      python_version: &DOCKER_PYTHON_VERSION py3
+      python_version: &DOCKER_PYTHON_VERSION py37
```
Comment (saimidu) [inline tensorflow/buildspec-tf1.yml:42]: Please change this to `py3` to stay consistent with what already exists. We will make an organized effort to update folder structures in the future, where our PR will spend time upon iterating on any build/test failures that come up due to the folder-name changes.

File: tensorflow/buildspec-tf1.yml [hunk 4]
```diff
@@ -59,9 +59,9 @@
     BuildTensorflowGPUTrainPy3DockerImage:
       <<: *TRAINING_REPOSITORY
       build: &TENSORFLOW_GPU_TRAINING_PY3 false
-      image_size_baseline: 4899
+      image_size_baseline: 5975
       device_type: &DEVICE_TYPE gpu
-      python_version: &DOCKER_PYTHON_VERSION py3
+      python_version: &DOCKER_PYTHON_VERSION py37
```
Comment (saimidu) [inline tensorflow/buildspec-tf1.yml:64]: Please change this to `py3`.

File: tensorflow/buildspec-tf1.yml [hunk 5]
```diff
@@ -70,100 +70,21 @@
...
-        <<: *TRAINING_CONTEXT
     BuildTensorflowExampleGPUTrainPy3DockerImage:
       <<: *TRAINING_REPOSITORY
       build: &TENSORFLOW_GPU_TRAINING_PY3 false
-      image_size_baseline: 4899
+      image_size_baseline: 5975
       base_image_name: BuildTensorflowGPUTrainPy3DockerImage
       device_type: &DEVICE_TYPE gpu
-      python_version: &DOCKER_PYTHON_VERSION py3
+      python_version: &DOCKER_PYTHON_VERSION py37
```
Comment (saimidu) [inline tensorflow/buildspec-tf1.yml:87]: Please change this to `py3`.

File: tensorflow/training/docker/1.15.4/py36/Dockerfile.cpu [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+# Specify the location of module that contains the training logic for SageMaker
+# https://docs.aws.amazon.com/sagemaker/latest/dg/docker-container-environmental-variables-entrypoint.html
+ENV SAGEMAKER_TRAINING_MODULE=sagemaker_tensorflow_container.training:main
+
+# Define framework-related package sources
+ARG TF_URL=https://aws-tensorflow-binaries.s3-us-west-2.amazonaws.com/tensorflow/r1.15_aws/20201002-042937/cpu/tensorflow_cpu-1.15.4-cp36-cp36m-manylinux2010_x86_64.whl
+
+# The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
+# the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
+ARG SMDEBUG_VERSION=0.9.3
```
Comment (saimidu) [inline tensorflow/training/docker/1.15.4/py36/Dockerfile.cpu:30]: Could you please verify that smdebug 0.9.3 is the correct version to use for TF 1.15.4?

File: tensorflow/training/docker/1.15.4/py36/cu101/Dockerfile.gpu [hunk 7]
```diff
@@ -0,0 +1,204 @@
+FROM nvidia/cuda:10.1-base-ubuntu18.04
```
Comment (saimidu) [inline tensorflow/training/docker/1.15.4/py36/cu101/Dockerfile.gpu:1]: Why does 1.15.4 have cu101 while 1.15.3 has cu100?

File: tensorflow/training/docker/1.15.4/py37/cu101/Dockerfile.gpu [hunk 8]
```diff
@@ -0,0 +1,213 @@
+FROM nvidia/cuda:10.1-base-ubuntu18.04
```
Comment (saimidu) [inline tensorflow/training/docker/1.15.4/py37/cu101/Dockerfile.gpu:1]: Why does 1.15.4 have cu101 while 1.15.3 has cu100?

File: tensorflow/buildspec-tf1.yml (hunk 5 above)
Comment (Teng-xu) [reply tensorflow/buildspec-tf1.yml]: Updated

File: tensorflow/buildspec-tf1.yml (hunk 4 above)
Comment (Teng-xu) [reply tensorflow/buildspec-tf1.yml]: Updated

File: tensorflow/buildspec-tf1.yml (hunk 3 above)
Comment (Teng-xu) [reply tensorflow/buildspec-tf1.yml]: Updated

File: tensorflow/training/docker/1.15.4/py36/cu101/Dockerfile.gpu (hunk 7 above)
Comment (Teng-xu) [reply tensorflow/training/docker/1.15.4/py36/cu101/Dockerfile.gpu:1]: We updated the cuda version from 10.0 to 10.1 for 1.15.4

File: tensorflow/training/docker/1.15.4/py37/cu101/Dockerfile.gpu (hunk 8 above)
Comment (Teng-xu) [reply tensorflow/training/docker/1.15.4/py37/cu101/Dockerfile.gpu]: We updated the cuda version from 10.0 to 10.1 for 1.15.4

File: tensorflow/training/docker/1.15.4/py36/Dockerfile.cpu (hunk 6 above)
Comment (Teng-xu) [reply tensorflow/training/docker/1.15.4/py36/Dockerfile.cpu]: Updated to 0.9.4

**Pull request closed without merging**